SwiftGit2/Repository.swift

```diff
@@ -558,4 +558,137 @@ final public class Repository {
+		guard lookupBaseGitResult == GIT_OK.rawValue, let unwrapBaseCommit = unsafeBaseCommit else {
+			return Result.failure(NSError(gitError: lookupBaseGitResult, pointOfFailure: "git_commit_lookup"))
+		}
+		git_commit_free(unsafeBaseCommit)
+
+		return Result.success(unwrapBaseCommit)
+	}
+
+	private func diff(fromTree oldTree: OpaquePointer?,
+	                  toTree newTree: OpaquePointer?) -> Result<OpaquePointer, NSError> {
+		var unsafeDiff: OpaquePointer? = nil
+		let diffResult = git_diff_tree_to_tree(&unsafeDiff, self.pointer, oldTree, newTree, nil)
+		guard diffResult == GIT_OK.rawValue, let unwrapDiffResult = unsafeDiff else {
+			return Result.failure(NSError(gitError: diffResult, pointOfFailure: "git_diff_tree_to_tree"))
+		}
+
+		return Result.success(unwrapDiffResult)
+	}
+
+	private func tree(from commit: OpaquePointer) -> Result<OpaquePointer, NSError> {
+		var unsafeTree: OpaquePointer? = nil
+		let treeResult = git_commit_tree(&unsafeTree, commit)
+		guard treeResult == GIT_OK.rawValue, let unwrapTree = unsafeTree else {
+			return Result.failure(NSError(gitError: treeResult, pointOfFailure: "git_commit_tree"))
+		}
+		git_tree_free(unsafeTree)
+
+		return Result.success(unwrapTree)
+	}
+
+	private func processDiffDeltas(_ diffResult: OpaquePointer) -> Result<[Diff.Delta], NSError> {
+		typealias Delta = Diff.Delta
+		var returnDict = [Delta]()
+
+		let count = git_diff_num_deltas(diffResult)
+
+		for i in 0..<count {
+			let delta = git_diff_get_delta(diffResult, i)
+			let gitDiffDelta = Diff.Delta(_: (delta?.pointee)!)
+
+			returnDict.append(gitDiffDelta)
+
+			git_diff_free(OpaquePointer(delta))
+		}
+
+		let result = Result<[Diff.Delta], NSError>.success(returnDict)
+		return result
+	}
+
+	// MARK: - Status
+
+	public func status() -> Result<[StatusEntry], NSError> {
+		var returnArray = [StatusEntry]()
+
+		// Do this because GIT_STATUS_OPTIONS_INIT is unavailable in swift
+		let pointer = UnsafeMutablePointer<git_status_options>.allocate(capacity: 1)
+		let optionsResult = git_status_init_options(pointer, UInt32(GIT_STATUS_OPTIONS_VERSION))
+		guard optionsResult == GIT_OK.rawValue else {
+			return Result.failure(NSError(gitError: optionsResult, pointOfFailure: "git_status_init_options"))
+		}
+		var options = pointer.move()
+		pointer.deallocate(capacity: 1)
+
+		var unsafeStatus: OpaquePointer? = nil
+		let statusResult = git_status_list_new(&unsafeStatus, self.pointer, &options)
+		guard statusResult == GIT_OK.rawValue, let unwrapStatusResult = unsafeStatus else {
+			return Result.failure(NSError(gitError: statusResult, pointOfFailure: "git_status_list_new"))
+		}
+
+		let count = git_status_list_entrycount(unwrapStatusResult)
+
+		for i in 0..<count {
+			let s = git_status_byindex(unwrapStatusResult, i)
+			if s?.pointee.status.rawValue == GIT_STATUS_CURRENT.rawValue {
+				continue
+			}
+
+			let statusEntry = StatusEntry(from: (s?.pointee)!)
```

Comment:
Might as well do `s!.pointee` here instead of `(s?.pointee)!`.

Reply:
:+1: